### PR TITLE
Replace query resource with react-router history push. Refs STCOM-302

### PR DESCRIPTION
**Pull request by mkuklis**
This PR is an attempt to replace navigation based on query resource with the tools provided by react-router. It should fix issues found in STCOM-302. 

I'm pretty sure the naming in the Location HOC (and perhaps `withLocation` name) could be improved. I wonder if you can come up with a better naming. 

I primary wanted to see if we can get rid of query resource completely and replace it with the set of navigation helpers (perhaps these helpers could be moved up to stripes-core eventually).

### CODE REVIEW

File: src/ViewHoldingsRecord.js [hunk 1]
```diff
@@ -846,7 +844,9 @@ ViewHoldingsRecord.propTypes = {
     temporaryLocationQuery: PropTypes.object.isRequired,
   }),
   onCloseViewHoldingsRecord: PropTypes.func.isRequired,
+  updateLocation: PropTypes.func,
+  goTo: PropTypes.func,
```
Comment (VictorSoroka1) [inline src/ViewHoldingsRecord.js]: Should these new props be required? I found the same situation in other files where that props added. Please check other places too.

File: src/withLocation.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import omitBy from 'lodash/omitBy';
+import isNil from 'lodash/isNil';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { parse, stringify } from 'query-string';
+import { compose } from 'redux';
+import { withRouter } from 'react-router';
+
+function withLocation(WrappedComponent) {
```
Comment (VictorSoroka1) [inline src/withLocation.js]: @mkuklis, about your concern about the name. I was thinking about `withHistory`. But eventually, think that `withLocation` might be the more appropriate here as what it really provides is the way to change location.

**Comment by VictorSoroka1** [comment]
@mkuklis, looks great!

File: src/ViewHoldingsRecord.js (hunk 1 above)
Comment (mkuklis) [reply src/ViewHoldingsRecord.js]: Thanks for catching this @VictorSoroka1!

**Pull request merged**